Release notes 1.9 (#414)
okteto/docs: src/content/self-hosted/install/releases.mdx

id: releases
---

## 1.9.0
12 June, 2023

### Deprecation Notice

- Support for Kubernetes [1.23](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.23.md) has been deprecated and will be removed in the next release.

> - Removed support for dedicated tls secrets for each ingress. 

Reviewer: Does this require any other change in the documentation?

Author: i don't see the property tlsSecret was being documented
cc @pchico83

Reviewer 2: We delete the docs a few weeks ago. I would make it clearer that this is for ingress created for the okteto installation

Author: @pchico83 as you have more context of it, could you propose a change for this item description?

Reviewer 2: I would go with no documenting it in the release notes

Author: removed

Author: removed

### Features

> - Kubernetes supported versions: [1.26](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.26.md), [1.25](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.25.md) and [1.24](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md).

Reviewer 3: ```suggestion
- Added support for Kubernetes v1.26, now supporting: [1.26](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.26.md), [1.25](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.25.md) and [1.24](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md).
```

Would this capture the change?

- Kubernetes supported versions: [1.26](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.26.md), [1.25](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.25.md) and [1.24](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md).

> - Upgraded the Okteto CLI version to [2.16.1](https://github.com/okteto/okteto/releases/tag/2.16.1)

Reviewer: ```suggestion
- Upgraded the Okteto CLI version to [2.16.2](https://github.com/okteto/okteto/releases/tag/2.16.2)
```

We just released `2.16.2` and it will be needed in the chart

> - Telemetry job is now launched as soon as the helm installation is completed.

Reviewer: Do you think this is relevant for customers to be included in the release notes @omnomagonz @pchico83

Reviewer 4: I don't see it as so relevant, but I might lack of context.

> - Upgrade ingress nginx to 4.5

Reviewer 2: Could we link to the nginx official release notes here?

> - Upgrade ingress nginx to 4.5

Author: ```suggestion
- Upgrade ingress-nginx to [4.5](https://github.com/kubernetes/ingress-nginx/releases/tag/controller-v1.6.4)
```

> - Admins can now check when it was the last interaction a user had with Okteto in the Admin dashboard, under the Users section.

Reviewer 5: ```suggestion
- Last user interaction with Okteto is now available at the Admin dashboard, under the Users section.
```

> - Admins can now check when it was the last interaction a user had with Okteto in the Admin dashboard, under the Users section.

Reviewer 3: ```suggestion
- Added a "Last Seen" column in the Admin Users dashboard so admins can see when their users last used Okteto
```

> - Remote deploys support when instance has self-signed certificates and private Certificate Authorities (CAs).

Reviewer 2: ```suggestion
- `okteto deploy --remote` supports when Okteto has self-signed certificates and private Certificate Authorities (CAs).
```

> - Now users have a new property `lastSeen` which is updated when the user interacts with Okteto. This property can be checked from the Admin dashboard, under the Users section.

Reviewer: ```suggestion
- Admins can now check when it was the last interaction a user had with Okteto in the Admin dashboard, under the Users section.
```

Reviewer: @rlamana @omnomagonz Should we update this doc section to include this information? https://www.okteto.com/docs/1.9/administration/dashboard/#users. Not blocking the release notes PR, just to track if it's needed or not that change

Reviewer 4: It does look like something worth mentioning in the docs. I will create the task in okteto/docs

> - Support to access Okteto on `127.0.0.1`

Reviewer: Do we have any documentation about how to configure this to link it here? cc @pchico83

Reviewer 2: This guide follows that approach:
https://www.okteto.com/docs/1.9/self-hosted/install/deployment/
But we don't have docs at the moment.

### Improvements

> - Frontend now loads faster with improved nginx compression

Reviewer 5: ```suggestion
- Enabled gzip compression for frontend web server
```

Reviewer 5: There were other changes that might impact loading times, so this improvement might not be perceived.

> - Frontend now loads faster with improved nginx compression

Reviewer 3: ```suggestion
- Faster load times on frontend app by improved nginx compression
```

BTW, could we consider removing the other _"now"_ usages in the PR? The "now" is implied through the change and new release, so I don't think such word usages really add any more value to the sentence? 

[More examples from past release notes](https://github.com/search?q=repo%3Aokteto%2Fdocs+now+path%3A%2F%5Esrc%5C%2Fcontent%5C%2Fself-hosted%5C%2Finstall%5C%2Freleases.mdx%2F&type=code) that similarly could be tidied up - Not in this PR, but calling it out for comparison and for future consideration.

cc @omnomagonz I'm wondering if we could consider documenting this and provide some guidelines/best-practices on how to articulate release notes more consistently in the future?

- Frontend now loads faster with improved nginx compression

> ### Bugfixes

Reviewer: @jpf-okteto Was the fix to prevent the error in the registry when there was an invalid manifest included in `1.9` or was it already on `1.8`?

- Improve UI static assets management across releases

> - Helm installations are now uninstalled always from dev environments on destroy all operations

Reviewer 3: ```suggestion
- Uninstall Helm installations from dev environments on "destroy all" operations
```

> - Improve UI static assets management across releases

Reviewer: How does this affect customers? Does this impact them somehow? What is fixing?

Author: As i read by the PRs this is an improvement at the UI

Reviewer 5: > How does this affect customers? Does this impact them somehow? What is fixing?

It does impact the users, as prior to this change, chunks were not always available across releases and could cause app to fail due to their unavailability.

Reviewer 5: ```suggestion
- Improve app chunk loading and error handling
```

Reviewer 3: @bgonp @tavferreira do you recall what this was about? I think we could/should articulate something more crisp - _What_ was _improved_? How? What _releases_? etc.

Reviewer 5: I think @teresaromero already merged my suggestion. Feel free to add any suggestions on your side @viddo

> - Any container in development mode is now deactivated as part of the sleep namespace operation

Reviewer 3: ```suggestion
- Deactivate any containers in development mode on "sleep namespace" operation
```

> - Helm installations are now uninstalled from dev environments when running destroying all at namespace and there is no `destroy` section at the manifest.

Reviewer: ```suggestion
- Helm installations are now uninstalled always from dev environments on destroy all operations
```

> - Variables defined at deploy time are now available on destroy operations

Reviewer 3: ```suggestion
- Environment variables defined for a deploy are available for destroy operation
```

> - Namespaces where not sleeping when any of its containers where in development mode (`okteto up`). Fixed so when a namespace is put to sleep al containers are deactivated.

Reviewer: ```suggestion
- Any container in development mode is now deactivated as part of the sleep namespace operation
```

> - Fixed a bug where the container image layer manifest could get corrupted upon being uploaded to the Registry. This would cause Buildkit to return an "500 Internal Server Error" for related build requests.

Reviewer 3: _"Fixed a bug"_ is implied by this being listed under _"Bugfixes"_ section already. Also, could this be articulated more clearly? Maybe something on these lines?

E.g. 
```suggestion
- Container image layer manifest sometimes got corrupted when being uploaded to the Builkdkit Registry, causing "500 Internal Server Error" on following build requests
```

> - Deploy Variables are now correctly saved at the git deploy object 

Reviewer: How does this affect to customers? I think we should specify something like

```
- Variables defined at deploy time are now available on destroy operations
```

Author: changed

> - Deploy Variables are now correctly saved at the git deploy object 

Author: ```suggestion
- Variables defined at deploy time are now available on destroy operations
```

id: releases
---

## 1.9.0
12 June, 2023

### Deprecation Notice

- Support for Kubernetes [1.23](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.23.md) has been deprecated and will be removed in the next release.
- Removed support for dedicated tls secrets for each ingress. 

> ### Features

Reviewer 2: @ifbyol should we mention any changes from lego epic 6 & 7'

Reviewer: We should mention that `okteto deploy --remote` is supported now in scenarios with self-signed certificates and private CAs. This is the main change introduced in this chart regrading epic 6.

For epic 7, I wouldn't mention anything as there is an scenario we need to review

- Kubernetes supported versions: [1.26](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.26.md), [1.25](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.25.md) and [1.24](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md).

> - Upgraded the Okteto CLI version to [2.15.4](https://github.com/okteto/okteto/releases/tag/2.15.4)

Reviewer 2: 2.16.1?

Author: yes, it was a mistake

>   - Fixed remote deploy using UI and adding variables

Author: The version of CLI was mistaken, 1.9 will have 2.16.1.

the tab list are worth-mentioning improvements from the cli version we include at the release

- Telemetry job is now launched as soon as the helm installation is completed.
- Now users have a new property `lastSeen` which is updated when the user interacts with Okteto. This property can be checked from the Admin dashboard, under the Users section.
- Support to access Okteto on `127.0.0.1`
- Upgrade ingress nginx to 4.5 

### Improvements

- Frontend now loads faster with improved nginx compression

> - An error is now displayed at the UI when the browser is not able to download a chunk.

Reviewer 5: +1 ☝️

### Bugfixes

> - "Loading chunk failed" error is now fixed caused when the app tried to download a chunk that no longer exist because a new release was launched.

Reviewer 6: ```suggestion
- Improve UI static assets management across releases
```
